server/game/cards/attachments/03/needle.js

```diff
@@ -8,13 +8,18 @@ class Needle extends DrawCard {
 
         this.interrupt({
             when: {
-                onSacrificed: event => event.card === this.parent
+                onSacrificed: event => {
+                    if(event.card === this.parent) {
+                        this.parentCard = this.parent;
+                        return true;
+                    }
```

Comment:
Not a fan of it, but at least in the case of Bodyguard, it's still a necessary evil currently right?